lib/pharos/cluster_manager.rb

```diff
@@ -82,5 +82,22 @@ def apply_addons
     def disconnect
       ssh_manager.disconnect_all
     end
+
+    def format_hosts(hosts)
+      return hosts.join(' ') if logger.debug?
+      case hosts.size
+      when 0
+        '<unknown>'
+      when 1..3
+        hosts.join(' ')
+      else
+        grouped = hosts.group_by { |h| h.split('.').first(2).join('.') + '.*' }
```

Comment:
Not convinced that grouping by IPv4 `/16` is the right thing to do, but I don't know what the best way to deal with very large numbers of hosts would be, in terms of the CLI output... perhaps some kind of spinner-like thing tied into the parallel execution threadpool showing which nodes are currently processing, and how many failed etc.

Reply:
Probably it's all very irrelevant to the user. I don't think the IPs are interesting at all and if anything fails, it should probably just abort asap.

It could just say "on worker nodes".

Reply:
The more nodes you have, the more likely it is for something to just fail randomly. I think aborting ASAP on errors is actually the wrong thing to do.

Reply:
> It could just say "on worker nodes".

Yes, that's also a good option.

Reply:
> The more nodes you have, the more likely it is for something to just fail randomly. I think aborting ASAP on errors is actually the wrong thing to do.

Yes, it's most likely too risky to abort everything if one node throws an error.